### PR TITLE
components: make content moderation configurable

### LINKED ISSUE #1861: Redesign content moderation component for extensibility

### PR DESCRIPTION
**Pull request by slint**
- Closes #1861.
- Adds a new `RRM_CONTENT_MODERATION_HANDLERS` config variable to allow
  for configuring multiple handlers for the different write actions.

### CODE REVIEW

**Review by ntarocco — APPROVED** [review]

File: invenio_rdm_records/services/components/verified.py [hunk 1]
```diff
@@ -27,4 +43,30 @@ def publish(self, identity, draft=None, record=None):
...
+        def _method(self, *args, **kwargs):
+            handlers = current_app.config.get(
+                "RRM_CONTENT_MODERATION_HANDLERS", {}
+            ).get(action, [])
+            for handler in handlers:
+                handler(*args, **kwargs)
+
+        return _method
+
+    update_draft = handler_for("update_draft")
```
Comment (ntarocco) [inline invenio_rdm_records/services/components/verified.py:67]: Tests are failing, I think you need:
```suggestion
    update_draft = ContentModerationComponent.handler_for("update_draft")
```

File: invenio_rdm_records/config.py [hunk 2]
```diff
@@ -555,6 +557,17 @@ def make_doi(prefix, record):
    def lock_edit_published_files(service, identity, record=None, draft=None):
 """
 
+RRM_CONTENT_MODERATION_HANDLERS = {
```
Comment (utnapischtim) [inline invenio_rdm_records/config.py:560]: question: is the `RRM` intended or a typo?

File: invenio_rdm_records/services/components/verified.py [hunk 3]
```diff
@@ -9,15 +9,31 @@
...
+from invenio_records_resources.services.uow import TaskOp
 from invenio_requests.tasks import request_moderation
 
 
-class ContentModerationComponent(ServiceComponent):
-    """Service component for content moderation."""
+class BaseHandler:
+    """Base class for content moderation handlers."""
+
+    def __init__(self, uow):
```
Comment (utnapischtim) [inline invenio_rdm_records/services/components/verified.py:19]: minor: uow should have a default value `= None`. python3.12 is complaining that in the config the parameter is not given

File: invenio_rdm_records/config.py (hunk 2 above)
Comment (ntarocco) [reply invenio_rdm_records/config.py:560]: Nice catch, totally missed this one!

File: invenio_rdm_records/services/components/verified.py (hunk 3 above)
Comment (slint) [reply invenio_rdm_records/services/components/verified.py:19]: In the end I refactored this to pass the `uow` as part of each handler (see after I push)

**Pull request merged**